tools/halide_image.h

```diff
@@ -148,9 +173,27 @@ class Image {
         contents->dev_free();
     }
 
-    Image(T vals[]) {
-        initialize(sizeof(vals)/sizeof(T));
-        for (int i = 0; i < sizeof(vals); i++) (*this)(i) = vals[i];
+    // Initialize the Image from a statically sized array.
+    // Because of the scalar overload of dimension_sizes, this will only work if Array
+    // is T or T const of one or more dimensions.
+    template< typename Array, size_t N >
```

Comment:
We always copy the data from vals into our own host storage (rather than keeping a reference); this should be made explicit in the comments. Also, since 'vals' are input only, it would probably be good to explicitly make it a const argument.